fineract-provider/src/main/java/org/apache/fineract/cob/service/ConfigJobParameterServiceImpl.java

```diff
@@ -59,15 +60,22 @@ public JobBusinessStepConfigData getBusinessStepConfigByJobName(String jobName)
     public CommandProcessingResult updateStepConfigByJobName(JsonCommand command, String jobName)
             throws BusinessStepNotBelongsToJobException {
         List<BusinessStep> businessSteps = dataParser.parseUpdate(command);
-        List<BatchBusinessStep> batchBusinessSteps = batchBusinessStepRepository.findAllByJobName(jobName);
-        businessSteps.forEach(newBusinessStepConfig -> {
-            BatchBusinessStep batchBusinessStep = batchBusinessSteps.stream()
-                    .filter(oldBusinessStepConfig -> oldBusinessStepConfig.getStepName().equals(newBusinessStepConfig.getStepName()))
-                    .findFirst().orElseThrow(BusinessStepNotBelongsToJobException::new);
-            batchBusinessStep.setStepName(newBusinessStepConfig.getStepName());
-            batchBusinessStep.setStepOrder(newBusinessStepConfig.getOrder());
-            batchBusinessStepRepository.save(batchBusinessStep);
-        });
+        List<String> availableBusinessStepNames = getAvailableBusinessStepsByJobName(BusinessStepCategory.LOAN.name())
```

Comment:
Performance improvement: You make this class to implement InitializingBean. In the afterPropertiesSet method can call the "getAvailableBusinessStepsByJobName" method and save the response at class level. The (application wise) available business step implementation will not be changed during runtime.